### DISCUSSION TITLE
Connection error to converter

### DISCUSSION THREAD

**Discussion by AndaleR** [nathanmarlor/foxess_modbus#612] [Support] [answered]
I finally bought an Elfi-part to get the modbus-informations with WIFI into HA.

Works and I have seen data coming. But: It crashes many times after some seconds / minutes?

The error-message in Home Assistant then is: `Error connecting to inverter (friendly name: {friendly_name}). Error reading registers. Type: RegisterType.HOLDING; start: 31000; count: 52; slave: 247 from tcp://192.168.179.51:502: Modbus Error: [Input/Output] No Response received from the remote slave/Unable to decode response`

Where do I have to look for?

I have made the settings as it should be for the Elfi (in the wiki). Have tried some different versions of the HA-foxess-modbus-module, similar errors.

**Comment by AndaleR** [comment]
Restarted everything new - now the error-message was: `Error connecting to inverter (friendly name: {friendly_name}). Error reading registers. Type: RegisterType.HOLDING; start: 41010; count: 1; slave: 247 from tcp://192.168.179.51:502: Modbus Error: [Input/Output] Modbus Error: [Invalid Message] No response received, expected at least 8 bytes (0 received)`
Took maybe 2-3 minutes?

**Comment by FozzieUK** [comment]
Those errors both report no response received - if the integration has been working then it sounds like the inverter & integration configuration is correct - in which case it's most likely the connection to the Elfin is failing, or there is a cabling problem.

Other users have reported with the Elfin power supply issues, i.e. it needs a good strong power supply to maintain it correctly (capable of a constant 1A) and also updating to latest firmware has helped.

On the cabling it should be twisted pair or CAT 5/6 cabling is it is running over distance (for a couple of meters it won't matter).

Also check your wifi signal strength is good and reliable where the Elfin is located.

**Comment by AndaleR** [comment, marked as answer]
Looks like it really was only a problem at the local network... Added a repeater closer to the Elfin - since that it works fine.

**Reply by canton7** [reply]
Thanks for letting us know!

**Reply by AndaleR** [reply]
How many times the informations are updated? As far as I remember the original App or the cloud-version for Foxess is doing it all 5 minutes.

How many times is the update-cycle of the modbus-addon?

**Reply by canton7** [reply]
We poll the inverter every 10 seconds. You can change this if you need to, under "CONFIGURE".

**Reply by AndaleR** [reply]
Thanks!
